### PR TITLE
Documentation: update iproute2 git URL in bpf.rst

### PR DESCRIPTION
**Pull request by dmitris**
PR fixes the git URL for iproute2 on https://docs.cilium.io/en/latest/bpf/#compiling-iproute2 - the current one doesn't work:
```
$ git clone git://git.kernel.org/pub/scm/linux/kernel/git/iproute2/iproute2.git /tmp/iproute
Cloning into '/tmp/iproute'...
fatal: remote error: access denied or repository not exported: /pub/scm/linux/kernel/git/iproute2/iproute2.git
```
Changed it to the https:// URL from the bottom of https://git.kernel.org/pub/scm/network/iproute2/iproute2.git which works fine.

### CODE REVIEW

**Review by pchaigno — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]
Thank you!

**Comment by dmitris** [comment]
sorry, @qmonnet  could you hold off on a merge for a bit? (Need to complete an internal approval procedure on my side since it's my first PR to this project and repository - expect to be back with a "go ahead" shortly). Thanks.

**Comment by dmitris** [comment]
all clear now - please go ahead and merge! Thanks. @qmonnet

**Pull request merged**
